hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/index/IndexResolver.java

```diff
@@ -135,21 +159,122 @@ private IndexResolver() {
             return Collections.emptyList();
         }
 
-        List<RelNode> rels = new ArrayList<>(supportedIndexes.size());
 
         for (MapTableIndex index : supportedIndexes) {
             // Create index scan based on candidates, if possible. Candidates could be merged into more complex
             // filters whenever possible.
-            RelNode rel = createIndexScan(scan, distribution, index, conjunctions, candidates);
 
-            if (rel != null) {
-                rels.add(rel);
+            RelNode relAscending = createIndexScan(scan, distribution, index, conjunctions, candidates, false);
+            RelNode relDescending = createIndexScan(scan, distribution, index, conjunctions, candidates, true);
```

Comment:
Index resolution is a heavy operation. Instead of calling it twice, we may just create a copy of `relAscending` and change the collation.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18013/commits/88f7e912313d4cb5d366a8934d311d3dfa05d50c